_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,70 @@
+Developers develop processes and methods. They align on conventions. And other developers violate those. Often times even consceniously. How to handle those violations?
+Often times now is the time when the reviewer changes from consultant to finger-pointing judge (and hangman) who points at the wiki in which he documented the rule three years ago (without ever discussing it properly). Usually the tone gets rough (catch some breath, adjust your mindset...).
+This behaviour is bad because creative people have the healthy habit to question rules if they are in their way. If the result is that the rule does not create value (but supresses value-creation) in the case the engineer, striving to achieve a goal, will try to navigate around it. In such a case: question the rule, not the violator (in fact violating/questioning rules is a healthy habit).
+So back to the situation. State your feedback like this: "at some point we decided to approach things like this like that because we were convinced thats better because of reason a and b. dont you agree? shall we re-evaluate?"
+Usually the topic is done now. The reviewee will probably accept the suggestion because the reviewer acted in a consulting role and took the reviewee serious, asked for his opinion.
+The reviewee will supposely not feel that the rule harms him so badly any more and follow it. Or a healthy discussion arises that in the end might change something for the better for the whole team.
+As a general rule of thumb: do not define too many rules and processes. And in a review: start with understanding the requirements, thinking about potential solutions also from a high-level/architectural/design-perspective. How would you have solved the problem? Then look at the solution. Find real bugs. Only beginners focus on cosmetics (and usually miss real flaws).
+
+# Treat cosmetics as cosmetics
+
+Cosmetics are called cosmetics because they are not substantial. Bad formatting, uncomplete documentation, bad naming or other trivial things should be handled as what they are: cosmetics. Dont ignore them, good make-up can make a difference. But as a reviewer, double-check to make sure your attitude fits the importance of the advice. Nothing is as bad as bad vibes because of things that do not make a real difference in the end.
+
+# Contribute but dont override
+
+It is generally not a bad idea to fix small issues during the code review. You can still point the reviewee at the changes but it does not make sense to invest the time twice. If you can fix a spelling error on the fly - why not do it?
+But dont change the design completely. Do not override the substance of what your reviewee wanted to do. Help him, dont control him.
+
+
+
+I hope I helped you to gain another angle on code reviews or at least made you reflect your review-patterns and opinions and maybe even helped you to improve.
+
+Best H
```

Comment:
I would like to keep the blog posts consistent and don't give just to your posts, the feeling of an email. 
In general, the personality of the writer heavily transpires through the way the article is written and I don't think having this, only for you is needed. Adding you're initial at the end, when right below it you will have your photo and your full name, makes it slightly redundant.